docs/sphinx/developers/code-formatting.txt

```diff
@@ -0,0 +1,30 @@
+Code formatting
+===============
+
+Note, these guidelines do not cover:
+
+- third party code imported into the source tree, which is covered by
+  the guidelines for the upstream projects
+- released schema files which would require re-releasing if changed by
+  reindenting
+
+All languages
+-------------
+
+- Use spaces to indent; do not ever use tabs
+
+Java
+----
+
+All Java code is formatted with:
+
+- an indentation size of two spaces
+- braces use the Java variant of K&R style
```

Comment:
any chance of an informative link/example? Java newbies may not have even heard of K&R